presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -158,6 +158,43 @@ public void testParsingError()
         assertQueryFails("SELECT foo FROM", "line 1:16: mismatched input '<EOF>'. Expecting: .*");
     }
 
+    @Test
+    public void testEmptyTableName()
+    {
+        //assertQuery("SELECT * FROM \" \"");
+        assertQueryFails("SELECT * FROM \"\"", "line 1:15: Tablename is empty");
+    }
+
+    @Test
+    public void testNullTableName()
+    {
+        assertQueryFails("SELECT * FROM NULL", "line 1:15: mismatched input 'null'. Expecting: .*");
+    }
+
+    @Test
+    public void testEmptySchemaName()
+    {
+        assertQueryFails("SELECT * FROM \"\".foo", "schemaName is empty");
```

Comment:
Where does this error come from?

Reply:
From SchemaUtil. Looking at adding a check in the visitTable function similar to the check for table name.